stl/inc/xutility

```diff
@@ -4805,12 +4805,32 @@ _INLINE_VAR constexpr bool _Fill_memset_is_safe = conjunction_v<is_scalar<_Ty>,
 template <class _FwdIt, class _Ty>
 _INLINE_VAR constexpr bool _Fill_memset_is_safe<_FwdIt, _Ty, false> = false;
 
+template <class _FwdIt, class _Ty, bool = is_pointer_v<_FwdIt>>
+_INLINE_VAR constexpr bool _Fill_zero_memset_is_safe =
+    conjunction_v<is_scalar<_Ty>, is_scalar<_Iter_value_t<_FwdIt>>, negation<is_member_pointer<_Iter_value_t<_FwdIt>>>,
+        negation<is_volatile<remove_reference_t<_Iter_ref_t<_FwdIt>>>>, is_assignable<_Iter_ref_t<_FwdIt>, const _Ty&>>;
+
+template <class _FwdIt, class _Ty>
+_INLINE_VAR constexpr bool _Fill_zero_memset_is_safe<_FwdIt, _Ty, false> = false;
+
 template <class _DestTy, class _Ty>
 void _Fill_memset(_DestTy* const _Dest, const _Ty _Val, const size_t _Count) {
     _DestTy _Dest_val = _Val; // implicitly convert (a cast would suppress warnings); also handles _DestTy being bool
     _CSTD memset(_Dest, static_cast<unsigned char>(_Dest_val), _Count);
 }
 
+template <class _DestTy>
+void _Fill_zero_memset(_DestTy* const _Dest, const size_t _Count) {
+    _CSTD memset(_Dest, 0, _Count * sizeof(_DestTy));
+}
+
+template <class _Ty>
+bool _Is_scalar_zero(const _Ty& _Val) {
+    // checks if scalar type has all bits set to zero, avoids using user defined operator== for enums
+    _STL_INTERNAL_STATIC_ASSERT(is_scalar_v<_Ty>);

```

Comment:
I think we should also have an `_STL_INTERNAL_STATIC_ASSERT` for `!is_member_pointer_v`, since that part of `_Fill_zero_memset_is_safe` is especially important here.

Reply:
Done